src/routes.tsx

```diff
@@ -11,25 +11,41 @@ import { FlowRecapPage } from '@/pages/FlowRecapPage'
 import { LoadingScreen } from '@/components/LoadingScreen'
 import { AccessibilityPage } from './pages/AccessibilityPage'
 import { ShortcutTutorialPage } from '@/pages/ShortcutTutorialPage'
+import { DeviceLimitPage } from './pages/DeviceLimitPage'
 import { OnboardingUtils } from '@/lib/utils/onboarding'
 import { useDeepLinkAuth } from './hooks/useDeepLinkAuth'
 import { getCurrentWindow } from '@tauri-apps/api/window'
 import { useEffect } from 'react'
 import { useGlobalShortcut } from './hooks/useGlobalShortcut'
 import { error as logError } from '@tauri-apps/plugin-log'
+import { useDeviceRegistration } from './hooks/useDeviceRegistration'
+
 
 const ProtectedRoute = () => {
-  const { user, loading } = useAuth()
+  const { user, loading: authLoading } = useAuth()
+  const { isBlockedByDeviceLimit, retryDeviceRegistrationCheck } = useDeviceRegistration()
   const location = useLocation()
 
-  if (loading) {
+  // Add check on route change
+  useEffect(() => {
```

Comment:
This works, but we might be able to "listen" to device updates using supabases api. I'll need to check

Reply:
Would be more efficient than calling on each route change